src/lib/options.ts

```diff
@@ -21,7 +21,6 @@ const ValidationOptionsSchema = Type.Object({
   logOptionsErrors: Type.Optional(Type.Boolean()),
   _internalThrowOnAdditionalProperties: Type.Optional(
     Type.Boolean({
-      default: process.env.NODE_ENV === "test",
```

Comment:
Also had to remove this to get `process.env` off the main code path. Was pretty trivial to override the property in `testYf.ts`, though

Reply:
As per above, I'll have some more comments on `process.env` further below.